Slim/Plugin/UPnP/MediaRenderer/ProtocolHandler.pm

```diff
@@ -66,11 +118,37 @@ sub isRepeatingStream {
 
 sub getMetadataFor {
 	my ( $class, $client, $url ) = @_;
-	
+
+	# This returns metadata for any tracks added to the playlist via the plugin.
+	# It can only return metadata for CurrentURI and NextURI so the playlist
+	# needs to be managed to only show those tracks. Any extra tracks will have
+	# the metadata for CurrentURI returned.
+
+	# $url =~ s/^http/upnp/; is returning an empty string for some reason so 
+	# stripping upnp/http from url so it can be compared.
+	my $strippedUrl = substr $url, 4;
 	my $pd   = $client->pluginData();
 	my $meta = $pd->{avt_AVTransportURIMetaData_hash};
 	my $res  = $meta->{res};
-	
+	my $currentUri = substr $res->{uri}, 4;
+
+	# if $url doesn't match CurrentURI check against NextUri
+	if( $strippedUrl ne $currentUri){
```

Comment:
Just a style nitpick: please add the space between the `if` keyword and `(`. And if you use a space inside the brackets, use them in the front and the end (which we usually do with lengthy conditions):

```perl
if ($strippedUrl ne $currentUri) {
...
```